packages/react-storage/src/components/StorageBrowser/views/Controls/Table.tsx

```diff
@@ -185,6 +217,15 @@ export const LocationDetailViewTable = ({
       ? pagedData.sort((a, b) => compareFn(a[selection], b[selection]))
       : pagedData.sort((a, b) => compareFn(b[selection], a[selection]));
 
+  // Selection state:
+  const files = pagedData.filter((item) => item.type === 'FILE');
+  const selectedMap: Record<string, boolean> = {};
+  selected.items?.forEach((item) => (selectedMap[item.key] = true));
+  const hasFilesToSelect = files.length > 0;
+  const allFilesSelected = hasFilesToSelect
+    ? files.every((item) => selectedMap[item.key])
+    : false;
```

Comment:
A bit lost here, what is this code used for?

Reply:
This is the logic for the "select all" checkbox in the table header

Reply:
Go it. Can we simplify the logic here a bit for readability? Something like:
```ts
const allFiles = pageData.filter(item => item.type === 'FILE');
const areAllFilesSelected = selected.length === allFiles.length;
const hasSelectableFiles = !!allFiles.length;
```